tests/lib/collectors/gcloud_e2e.py

```diff
@@ -109,28 +88,42 @@ def test_end_to_end_boot_disk(self):
       analysis purpose and attach the boot disk copy to it.
     """
 
-    self.test_state = state.DFTimewolfState(config.Config)
-    gcloud_collector = gcloud.GoogleCloudCollector(self.test_state)
-    self.assertIsNotNone(gcloud_collector)
-
-    # This should make a copy of the boot disk only
-    gcloud_collector.SetUp(
+    # Setting up the collector to make a copy of the boot disk only
+    self.gcloud_collector.SetUp(
         self.project_id,
         self.project_id,
         self.incident_id,
         self.zone,
         42.0,
         16,
         remote_instance_name=self.instance_to_analyse,
+        # disk_names=None by default, boot disk will be copied
     )
 
-    self.analysis_vm = gcloud_collector.analysis_vm
-    self.assertEqual(gcloud_collector.disk_names, [])
-    self.__assertCollectorIsInitialized(gcloud_collector)
-
     # Attach the boot disk copy to the analysis VM
-    gcloud_collector.Process()
-    self.__assertInstanceAndDisk()
+    self.gcloud_collector.Process()
+
+    # The forensic instance should be live in the analysis GCP project and
+    # the disk should be attached
+    analysis_vm_name = self.test_state.output[0][0]
+    attached_disk_name = self.test_state.output[0][1].name
+    try:
+      operation = self.gcp.gce_api().instances().get(
+          project=self.project_id,
+          zone=self.zone,
+          instance=analysis_vm_name).execute()
+      result = self.gcp.gce_operation(operation, zone=self.zone)
+
+      self.assertEqual(result['name'], analysis_vm_name)
+      for disk in result['disks']:
+        if disk['source'].split("/")[-1] == attached_disk_name:
+          return
+      self.fail('Error: could not find the disk {0:s} in instance {1:s}'.format(
+          attached_disk_name, analysis_vm_name
+      ))
+    except RuntimeError as exception:
```

Comment:
What in the code is expected to raise a `RuntimeError`? Make sure  to narrow down the try / except blocks as much as possible. Also, it might be fine to just not try to catch this exception as we'll get a full stacktrace on where the error came from (easier for debugging, which is what tests are for)

Reply:
The calls to the GCP API done by the `gcp` module would raise a RuntimeError if the API call fails, but I agree this can be discarded.